client/ayon_core/hosts/nuke/plugins/publish/collect_headless_farm.py

```diff
@@ -0,0 +1,41 @@
+import pyblish.api
+
+
+class CollectHeadlessFarm(pyblish.api.InstancePlugin):
+    """Setup instances for headless farm submission."""
+
+    order = pyblish.api.CollectorOrder + 0.4999
+    label = "Collect Headless Farm"
+    hosts = ["nuke"]
+
+    def process(self, instance):
```

Comment:
Does this have to be instance plugin?

Reply:
We need to iterate over the instances to find the right and also the workfile instance. Could be context plugin, but we would be needing to iterate anyways.
Why should it not be instance plugin?

Reply:
The plugin does not have any instance filters. Making in context plugin would make it run once. You can actually validate if there is any instance matching the node name, current implementation does not validate it in any way, just turns off all that don't match the conditions, but don't validate there is one which does match.

Reply:
Done